### PR TITLE
chore: combine coverage reports from all platforms supported

### PR DESCRIPTION
**Pull request by jnoortheen**
also upgrade codecov-action version

* internal change: no news item

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

File: .github/workflows/pytest-windows-3.9.yml [hunk 1]
```diff
@@ -55,6 +55,11 @@ jobs:
         run: |
           python -m pip --version
           python -m pip install -r requirements/tests.txt
-          python -m pip install . --no-deps
+          python -m pip install -e . --no-deps
       - name: Run tests
-        run: python -m xonsh run-tests.xsh test -- --timeout=240
+        run: python -m xonsh run-tests.xsh test --report-coverage --no-amalgam -- --timeout=240
```
Comment (daniel-shimon) [inline .github/workflows/pytest-windows-3.9.yml:60]: Why `--no-amalgam`? These tests need to run with the amalgamated version to prevent bugs like #4289

**Review by daniel-shimon — CHANGES_REQUESTED** [review]

File: .github/workflows/pytest-windows-3.9.yml (hunk 1 above)
Comment (jnoortheen) [reply .github/workflows/pytest-windows-3.9.yml:60]: these measure coverage and need actual code.

Comment (jnoortheen) [reply .github/workflows/pytest-windows-3.9.yml:60]: There are other tests running with amalgam

Comment (daniel-shimon) [reply .github/workflows/pytest-windows-3.9.yml:60]: So now only 3.9 tests run without amalgamation?

Comment (jnoortheen) [reply .github/workflows/pytest-windows-3.9.yml:60]: Yes. now only linux+3.9 is used for getting coverage

**Review by daniel-shimon — APPROVED** [review]
Cool! Thanks

**Pull request merged**